[//]: # (werk v2)
# agent_netapp_ontap: KeyError: 'used'

key        | value
---------- | ---
date       | 2024-10-09T16:23:37+00:00
version    | 2.4.0b1
class      | fix
edition    | cre
component  | checks
level      | 1
compatible | yes

This change affects users monitoring NetApp through the Ontap REST API. Previously
the special agent might crash with the following error
```
 File "/omd/sites/IDT_Dessau/lib/python3/cmk/special_agents/agent_netapp_ontap.py", line 287, in fetch_luns
    space_used=element_data["space"]["used"],
KeyError: 'used
```
If this happens now, the state of the check will be 'UNKN' and the summary will contain a suggestion that the space used is not available.

It is important to note that we have not been able to reproduce the problem where the lun volume does not provide information about how much space is used.
Therefore, other errors may occur in this scenario that cannot be addressed by this werk. Please report such problems so that we can address them.
